File: Server/src/main/java/com/api/server/foodb/CiteThisArticle.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "cite_this_articles")
public class CiteThisArticle {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "pubmed_id", nullable = false)
    private Integer pubmedId;

    @Lob
    @Column(name = "citation")
    private String citation;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Size(max = 255)
    @Column(name = "ref_id")
    private String refId;

    @Size(max = 255)
    @Column(name = "doi")
    private String doi;

    @Size(max = 1000)
    @NotNull
    @Column(name = "title", nullable = false, length = 1000)
    private String title;

    @NotNull
    @Lob
    @Column(name = "authors", nullable = false)
    private String authors;

    @Size(max = 255)
    @Column(name = "source")
    private String source;

    @Size(max = 255)
    @Column(name = "journal")
    private String journal;

    @Size(max = 255)
    @Column(name = "volume")
    private String volume;

    @NotNull
    @Column(name = "year", nullable = false)
    private Integer year;

    @Size(max = 255)
    @NotNull
    @Column(name = "date", nullable = false)
    private String date;

    @Size(max = 255)
    @Column(name = "pages")
    private String pages;

    @Size(max = 255)
    @Column(name = "issue")
    private String issue;

    @Lob
    @Column(name = "abstract")
    private String abstractField;

}

File: DotNetServer/DotNetServer/Entities/CiteThisArticle.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Entities;

[Table("cite_this_articles")]
[Index("PubmedId", Name = "index_cite_this_articles_on_pubmed_id", IsUnique = true)]
public partial class CiteThisArticle
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("pubmed_id")]
    public int PubmedId { get; set; }

    [Column("citation", TypeName = "text")]
    public string? Citation { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [Column("ref_id")]
    [StringLength(255)]
    public string? RefId { get; set; }

    [Column("doi")]
    [StringLength(255)]
    public string? Doi { get; set; }

    [Column("title")]
    [StringLength(1000)]
    public string Title { get; set; } = null!;

    [Column("authors", TypeName = "text")]
    public string Authors { get; set; } = null!;

    [Column("source")]
    [StringLength(255)]
    public string? Source { get; set; }

    [Column("journal")]
    [StringLength(255)]
    public string? Journal { get; set; }

    [Column("volume")]
    [StringLength(255)]
    public string? Volume { get; set; }

    [Column("year")]
    public int Year { get; set; }

    [Column("date")]
    [StringLength(255)]
    public string Date { get; set; } = null!;

    [Column("pages")]
    [StringLength(255)]
    public string? Pages { get; set; }

    [Column("issue")]
    [StringLength(255)]
    public string? Issue { get; set; }

    [Column("abstract", TypeName = "text")]
    public string? Abstract { get; set; }

    [InverseProperty("Article")]
    public virtual ICollection<CiteThisArticleReferencing> CiteThisArticleReferencings { get; set; } = new List<CiteThisArticleReferencing>();
}
